### ISSUE TITLE
Panic when trying to run spotifyd 0.3.2

### ISSUE THREAD

**Issue by LukeMarlin** [Spotifyd/spotifyd#1027]
**Description**
Cannot start spotifyd due to:
>Caught panic with message: called `Result::unwrap()` on an `Err` value: DNSError(Unknown)
shutdown; state=pool::State { lifecycle: Running, num_futures: 0 }

**To Reproduce**
1. Install Spotifyd from Arch User Repository
2. Do a config file with only username and password
3. Run `spotifyd --no-daemon --verbose`

**Expected behavior**
Spotifyd starts

**Logs**
<details><summary>Click to show logs</summary>

```
spotifyd --no-daemon --verbose
Loading config from "/home/x/.config/spotifyd/spotifyd.conf"
CliConfig { config_path: None, no_daemon: true, verbose: true, pid: None, shared_config: SharedConfigValues { username: Some("taken out for privacy"), username_cmd: None, password: Some("taken out for privacy"), password_cmd: None, use_keyring: false, use_mpris: None, on_song_change_hook: None, cache_path: None, no-audio-cache: false, backend: None, volume_controller: None, device: Some("Spotify-tui-tour"), control: None, mixer: None, device_name: None, bitrate: None, initial_volume: None, volume_normalisation: false, normalisation_pregain: None, zeroconf_port: None, proxy: None, device_type: None } }
Found user shell: Some("/bin/bash")
No proxy specified
registering with poller
registering with poller
registering with poller
registering with poller
build; num-workers=4
registering with poller
Using software volume controller.
starting background reactor
adding I/O source: 0
registering with poller
Zeroconf server listening on 0.0.0.0:34043
*** WARNING *** The program 'spotifyd' uses the Apple Bonjour compatibility layer of Avahi.
*** WARNING *** Please fix your application to use the native API of Avahi!
*** WARNING *** For more information see <http://0pointer.de/blog/projects/avahi-compat.html>
Caught panic with message: called `Result::unwrap()` on an `Err` value: DNSError(Unknown)
shutdown; state=pool::State { lifecycle: Running, num_futures: 0 }
  -> transitioned to shutdown
  -> shutting down workers
deregistering handle with poller
```

</details>

**Compilation flags** 
- [x] dbus_mpris
- [x] dbus_keyring
- [ ] alsa_backend
- [ ] portaudio_backend
- [x] pulseaudio_backend
- [x] rodio_backend

**Versions (please complete the following information):**
- OS: Archlinux
- Spotifyd: 0.3.2
- cargo: 1.57.0

**Comment by eladyn** [comment]
Hey!

Just as an idea, I found an [issue in the librespot repository](https://github.com/librespot-org/librespot/issues/819), which describes something very similar. I don't think that spotifyd enables `with-dns-sd` by default and thus this could be a completely different problem, but it might still help you.

You could also try if [running bare `librespot`](https://github.com/librespot-org/librespot/tree/v0.2.0#quick-start) works. It might very well be a problem on their side.

**Comment by LukeMarlin** [comment]
```
$ librespot -n "Librespot Speaker" -b 160
[2021-12-19T22:58:24Z INFO  librespot] librespot 0.3.1 UNKNOWN (Built on 2021-12-19, Build ID: 6JaKKo5Y, Profile: release)
```

Librespot works out of the box (cargo install). It is also fine if I compile it myself.

**Comment by eladyn** [comment]
I assume that it should have been fixed with 0.3.3. If that isn't the case, feel free to reopen. :)
